### ISSUE TITLE
Poster on IPFS & IPLD at Hydra Connect 2016 in Boston

### ISSUE THREAD

**Issue by flyingzumwalt** [ipfs/community#173]
@flyingzumwalt and @RichardLitt will be attending [Hydra Connect](https://wiki.duraspace.org/display/hydra/Hydra+Connect+2016) in Boston and presenting a poster on IPFS, authenticated data structures, IPLD and their applicability to libraries & archives.  Is anyone else interested in coming along? @nicola would you be interested in coming along to field questions about IPLD?

**The Audience** is _Libraries & Archives who use linked data._ The Hydra Framework stores all of its data as linked data using an LDP implementation (defaults to Apache Modeshape), so everyone at this conference knows about Linked Data and has valuable collections formatted as Linked Data. You'll have the full spread -- from people who just know Linked Data is a big deal to the people who designed the LDP implementation that Hydra relies on.

**When** Tuesday October 4th 2.00 PM - 5.00 PM
**Where** Boston Public Library

The conference is Oct 3-6, but the [poster session is 2:00-5:00 on Tuesday](https://wiki.duraspace.org/display/hydra/Hydra+Connect+2016#HydraConnect2016-Tuesday4thOctober(AtBostonPublicLibrary)).

**Comment by nicola** [comment]
Will be there! Ping me the night before we can organize an IPFS dinner at MIT

**Comment by chriscool** [comment]
Is it "Hydra Connect 2016" or "Hydra Connect 2017"?

**Comment by flyingzumwalt** [comment]
Oops. It's 2016. I was reading another ticket about a conference in 2017 right before I posted this. ☺️ I'll fix the title.

**Comment by flyingzumwalt** [comment]
@RichardLitt Let's use this ticket to track work on the poster. Here's an initial stab at the message(s) we should frame the poster around.  The audience is defined in the first comment above.

### Key Message
_What message do we want the audience to walk away with?_

IPFS and the decentralized web are the key to the future of Libraries and Archives in a data-centric world.  They provide a clear role for Libraries to fulfill their missions by directly engaging with scholarly, civic, and social data while applying models for preservation and access that are less expensive and infinitely scalable, and cryptographically authenticated.

### Secondary Messages
_What 3 messages do we want to follow up with?_

1. IPLD solves some of the core preservation problems that occur with Linked Data by using content-addressed authenticated data structures instead of semantic web and location-addressed http URIs.
2. IPFS covers the LOCKSS/DPN use cases gracefully with an open, stable, secure protocol that's used across many industries.
3. It's time to stop thinking in terms of Web Applications and REST APIs. Instead think of authenticated/functional data structures, p2p replication, and dynamic content on distributed systems.

**Comment by jbenet** [comment]
👍  LGTM

> "instead of semantic web"

only comment: I would probably not say "instead of semantic web", as people can write RDF -> JSON-LD -> IPLD, rather say the emphasis is lower level or something

**Comment by flyingzumwalt** [comment]
FYI: we need to [submit the poster to the printers in Boston](https://wiki.duraspace.org/display/hydra/Hydra+Connect+2016+Poster+sign-up+and+details+of+local+printing+arrangements#HydraConnect2016Postersign-upanddetailsoflocalprintingarrangements-Detailsoflocalprintingarrangements) by Monday 26 September.

**Comment by flyingzumwalt** [comment]
@RichardLitt your draft of the poster looks good.  Some notes:

**Title**:
I want the title to make a really bold statement along the lines of _IPFS and the decentralized web are the key to the future of Libraries and Archives in a data-centric world._

Note: This is not just about Archiving Data. Yes, they will use IPFS for archiving, but it also allows them to cover a lot of their core values.  Look at the American Library Association’s [Core Values of Librarianship](http://www.ala.org/advocacy/intfreedom/statementspols/corevalues).  The design of IPFS speaks to a majority of these values, only one of which is Preservation. Adopting IPFS will allow them to double down on their commitment to those values in a big way.

_Three Key Terms you want to have on the tip of your tongue: Preservation, Discovery and Access_. Any library or archive has to support all three of these activities in order to fulfill its mission. We want to speak to all three.

**“Problem”**:
Rename this top section to “The Promise” or “The Future”.  The problem, which I don’t think we should put on this poster, is that libraries are scrambling to define their role in the data-centric world. The solution is to adopt IPFS in a big way. That opens up wide avenues for a resurgence of Libraries 

* libraries, including small public libraries, pinning datasets their patrons care about
* university libraries seamlessly archiving complete histories of research datasets by simply allowing their stakeholders to submit the IPNS hashes of datasets that should be archived
* LOCKSS/DPN become dead simple to implement and support, barrier to entry becomes negligible and overhead for maintenance of the network is radically reduced
* Overhead for Hosting content plummets due to the bittorrent-style model for delivery
* Librarians get to resume their role of facilitators and curators who engage directly with the community to help them preserve and access content

**How IPFS Works**
We definitely need a picture of a Merkle DAG _with_ example hashes. Ideally there should be some visual representation that a single file is represented as a tree (not just one block) and a Directory is just another block that contains a name and a list of pointers to the hashes of the directory’s children. But if we can’t figure out a visual representation of that, let’s just emphasize a DAG with hashes.

**How IPLD Works**
Key idea: instead of relying on HTTP URIs to resolve links in the graph, use cryptographic hashes.  This makes the entire graph an _authenticated_, immutable dataset that 
* natively accommodates signing assertions (attribution/provenance of metadata)
* inherently supports replication/redundancy
* scales like bitTorrent
* inherently supports compact, efficient versioning

**Further Reading**
The ones you listed plus:
Our twitter handles
?: Should we include links to the incomplete [decentralized web primer](https://flyingzumwalt.gitbooks.io/decentralized-web-primer/content/) and [decentralized data workshops](https://codeforphilly.github.io/decentralized-data/)?

**Comment by RichardLitt** [comment]
Updated. How does this look? 

ipfs.io/ipfs/QmR2tpzoKvZheNBwEPGFWPsb6N2SQpCJntiZqJrk75KNu5

**Comment by RichardLitt** [comment]
Done! See the keypoint and PDF here. 

https://ipfs.io/ipfs/QmbkWejNBywf51afx8iiLXfL2U2q17ajqW5vLShy7JHeYx

**Comment by lgierth** [comment]
Looks pretty good to me! The footer still has a stub email address

**Comment by RichardLitt** [comment]
Good catch!

**Comment by nicola** [comment]
(off-topic: shall we host either a dinner at my house (if we are few) or drinks for IPFS peeps in Boston, the night before?)

**Comment by RichardLitt** [comment]
@nicola That sounds good to me! That's on-topic, I would say. When works for you? Maybe the 3rd, though; I arrive late on the 2nd.

**Comment by flyingzumwalt** [comment]
@nicola That sounds wonderful! Will you be able to come to the poster session? I'm trying to figure out the best time to let you converse with the people who designed their LDP implementation.

**Comment by sdellis** [comment]
❤️   I know I'm late to the game here, but I just want to express my excitement that you are all presenting this at Hydra Connect!  I won't be there, but colleagues of mine will, and it would be great for them to hear about IPFS from someone other than me! 😄   

And from my conversations at work, for us to even consider IPFS as a preservation alternative, any vendor of IPFS solutions will need to "guarantee and insure" the integrity of the preserved objects.  Apparently, our current storage vendor for our preservation file system does this, so that's a selling point to the "management" folks. 

Small point: I think the printing deadline has passed, but if not, perhaps "This is our vision of the future:" is stronger than
> This our imagining of the future:

**Comment by flyingzumwalt** [comment]
Thanks for the tip Shawn. I'm looking forward to it too.  Good point about the wording, but it's already been sent off to the printers.

**Comment by Kubuxu** [comment]
@sdellis the tool you are looking for is ipfs-cluster, unfortunately it doesn't exist yet but we are thinking about it. https://github.com/ipfs/notes/issues/58

**Comment by flyingzumwalt** [comment]
The poster was well received and @flyingzumwalt's lightning talk was very well received.

Notes for future presentations:
* Don't assume that people know how big IPFS is. One person thought that it was just a random idea some guy at a library wants all the libraries to use.
* Make sure to tell people about the big, active, exciting, innovative community, and tell them where to find it.
* The "thin waist" metaphor is really useful -- IPFS replaces the thin waist of the current web (IP addresses) with Merkle DAGs

**Comment by peterVG** [comment]
Great to see IPFS project reaching out to library and archives community like this (see also [https://github.com/ipfs/notes/issues/170]). As @sdellis points out above, IPFS-cluster (or something like it, e.g. pin whitelists) will be needed before any of these institutions should consider it for long-term preservation storage. Of course, IPFS for access copies is ready to rock today. 

The iPres digital preservation conference just wrapped up. Here is a working draft of digital preservation storage requirements that came out of one of the iPres workshops against which potential solutions, including IPFS, can be evaluated. https://docs.google.com/document/d/1WByfeonuTae5oOckSgeWeXI1TRntJE9RNMhMQqh4sS8/edit This set of requirements could help any future IPFS-for-preservation business cases and inform discussions about IPFS-cluster feature priorities.

**Comment by flyingzumwalt** [comment]
Thanks for sharing that Storage Criteria document @peterVG. That's super useful!

**Comment by RichardLitt** [comment]
This was great! Was super awesome to be at Hydra. @flyingzumwalt Anything we should add, before we close this?

**Comment by flyingzumwalt** [comment]
I think we're good. This issue kicked up some good discussions and I think we created issues for each of those discussions to continue. Closing this issue.
